src/cpp-common/include/cpp-common/HashMap.tpp

```diff
@@ -0,0 +1,145 @@
+#ifndef __HASH_MAP_TPP_
+#define __HASH_MAP_TPP_
+
+#include "HashMap.h"
+#include <cstdlib>
+#include <cstring>
+#include <new>
+#include <string>
+
+template <typename Key, typename MappedType, uint16_t maxSize>
```

Comment:
Add a TODO to use quadratic probing instead of linear probing, it's not needed for now, but ideally we could use quadratic probing